src/main/java/picard/vcf/GenotypeConcordance.java

```diff
@@ -457,70 +584,89 @@ public static void outputDetailMetricsFile(final VariantContext.Type variantType
-            // Note.  Genotype.isMixed means that it is called on one chromosome and NOT on the other
-            else if ((truthGenotype.isMixed())) truthState = TruthState.NO_CALL;
+        public List<Allele> truthAlleles() {
+            if (allAlleles.isEmpty() || this.truthAllele1 == null) {
+                return Collections.emptyList();
+            }
+            else {
+                final Allele truthAllele1 = Allele.create(this.truthAllele1, this.allAlleles.indexOf(this.truthAllele1) == 0);
+                final Allele truthAllele2 = Allele.create(this.truthAllele2, this.allAlleles.indexOf(this.truthAllele2) == 0);
+                return Arrays.asList(truthAllele1, truthAllele2);
+            }
         }
 
-        // Site level checks
-        if (callContext == null) callState = CallState.MISSING;
-        else if (callContext.isMixed()) callState = CallState.IS_MIXED;
-        else if (callContext.isFiltered()) callState = CallState.VC_FILTERED;
-        else {
-            // Genotype level checks
-            callGenotype = callContext.getGenotype(callSample);
-            if (callGenotype.isNoCall())           callState = CallState.NO_CALL;
-            else if (callGenotype.isFiltered())    callState = CallState.GT_FILTERED;
-            else if ((callGenotype.getGQ() != -1) && (callGenotype.getGQ() < minGq)) callState = CallState.LOW_GQ;
-            else if ((callGenotype.getDP() != -1) && (callGenotype.getDP() < minDp)) callState = CallState.LOW_DP;
-                // Note.  Genotype.isMixed means that it is called on one chromosome and NOT on the other
-            else if ((callGenotype.isMixed())) callState = CallState.NO_CALL;
+        public List<Allele> callAlleles() {
+            if (allAlleles.isEmpty() || this.callAllele1 == null) {
+                return Collections.emptyList();
+            }
+            else {
+                final Allele callAllele1 = Allele.create(this.callAllele1, this.allAlleles.indexOf(this.callAllele1) == 0);
+                final Allele callAllele2 = Allele.create(this.callAllele2, this.allAlleles.indexOf(this.callAllele2) == 0);
+                return Arrays.asList(callAllele1, callAllele2);
+            }
         }
+    }
+
+    /** Gets the alleles for the truth and call genotypes.  In particular, this handles the case where indels can have different
+     * reference alleles. */
+    final private static Alleles normalizeAlleles(final VariantContext truthContext,
+                                                  final String truthSample,
+                                                  final VariantContext callContext,
+                                                  final String callSample) {
+
+        final Genotype truthGenotype, callGenotype;
+
+        if (truthContext == null || truthContext.isMixed() || truthContext.isFiltered()) truthGenotype = null;
```

Comment:
I'm not sure if it's done elsewhere, but before calling `isMixed()` would it make sense to produce sub-contexts that are just for the sample of interest?  I.e. we don't much care if a second sample has an allele that makes the context mixed.